operations/commit_queue.go

```diff
@@ -117,3 +180,101 @@ func deleteCommitQueueItem(ctx context.Context, client client.Communicator, proj
 
 	return nil
 }
+
+type mergeParams struct {
+	projectID string
+	ref       string
+	id        string
+	pause     bool
+}
+
+func (p *mergeParams) mergeBranch(ctx context.Context, client client.Communicator, ac *legacyClient) error {
+	if p.id == "" {
+		if p.projectID == "" {
+			return errors.New("no project ID provided")
+		}
+
+		projectRef, err := ac.GetProjectRef(p.projectID)
+		if err != nil {
+			return err
+		}
+
+		diffData, err := getFeaturePatchInfo(projectRef.Branch, p.ref)
+		if err != nil {
+			return errors.Wrap(err, "can't generate patches")
+		}
+
+		if len(diffData.patches) == 0 {
+			if !confirm("Patch submission is empty. Continue?(y/n)", true) {
+				return nil
+			}
+		}
+
+		grip.Info(diffData.stat)
+		grip.Info(diffData.log)
+		if !confirm("This is a summary of the merge patch to be submitted. Continue? (y/n):", true) {
+			return nil
+		}
+
+		p.id, err = uploadPatches(ctx, client, p.projectID, diffData.patches)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !p.pause {
+		if err := enqueueItem(ctx, client, p.id); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type diffData struct {
+	stat    string
+	log     string
+	patches string
+}
+
+func getFeaturePatchInfo(projectBranch, ref string) (diffData, error) {
+	upstream := projectBranch + "@{upstream}"
```

Comment:
This assumes the user is branching off a local branch that a) has the same name as the project's tracked branch and b) is tracking the project's tracked branch.
This seems reasonable, considering we already assume this for patches.